File: Ayudantia 11/src/ayudantia/pkg11/ListaClientes.java
package ayudantia.pkg11;

/**
 *
 * @author Cristóbal
 */
public class ListaClientes { //UN NEXO

    private NodoCliente first;

    public ListaClientes() {
        this.first = null;
    }

    public NodoCliente getFirst() {
        return first;
    }

    public void setFirst(NodoCliente first) {
        this.first = first;
    }

    public void ingresarCliente(Cliente C) {//MÉTODO INGRESAR AL PRINICIPIO.
        NodoCliente nuevo = new NodoCliente(C);
        if (first == null) {
            first = nuevo;
        } else {
            nuevo.setNext(first);
            first = nuevo;
        }
    }

    /*public void ingresarCliente(Cliente C) {//MÉTODO INGRESAR AL FINAL.
        NodoCliente nuevo = new NodoCliente(C);
        if (first == null) {
            first = nuevo;
        } else {
            first.setNext(nuevo);
        }
    }*/
    
    public Cliente buscarCliente(String rut){
        NodoCliente actual = first;
        while(actual != null && !actual.getCliente().getRut().equalsIgnoreCase(rut)){
            actual = actual.getNext();
        }
        
        if(actual != null){
            return actual.getCliente();
        }else{
            return null;
        }
    }
    
    public boolean eliminarCliente(String rut){
        NodoCliente actual = first;
        NodoCliente prev = null;
        while(actual != null && !actual.getCliente().getRut().equalsIgnoreCase(rut)){
            actual = actual.getNext();
        }
        
        if(actual != null){
            if(first == null){
                first = first.getNext();
                return true;
            }else{
                prev.setNext(actual.getNext());
                return true;                
            }
        }else{
            return false;          
        }  
    }
    
}

File: Taller1ED_2018/ListaClientes.cpp

#include "ListaClientes.h"
#include <cstdlib>
#include <string.h>
#include <iostream>
using namespace std;

ListaClientes::ListaClientes() {
    this->first=NULL;
    this->last=NULL;
}

NodoCliente* ListaClientes::getFirst() {
    return first;
}

NodoCliente* ListaClientes::getLast() {
    return last;
}

void ListaClientes::setFirst(NodoCliente*) {
    this->first=first;
}

void ListaClientes::setLast(NodoCliente*) {
    this->last=last;
}

void ListaClientes::ingresarCliente(Cliente* C) {
    NodoCliente *nuevo = new NodoCliente(C);
    if(first==NULL) {
        first=nuevo;
        last=nuevo;
    }else{
        nuevo->setNext(first);
        first->setPrev(nuevo);
        nuevo->setPrev(NULL);
        first=nuevo;
    }

}

Cliente *ListaClientes::buscarCliente(string rut) {
    NodoCliente *current = first;
    while(current != NULL && current->getCliente()->getRut() != rut){
        current = current->getNext();
    }   
    if(current != NULL) {
        return current->getCliente();
    }else{
        return NULL;
    }

}

bool ListaClientes::eliminarCliente(string rut) {
    NodoCliente *current = first;
    while(current!= NULL && current->getCliente()->getRut() != rut) {
        current = current->getPrev();
    }
    if(current != NULL){
        if(current == first) {
            NodoCliente *eliminado = first;
            first = first->getNext();
            delete eliminado;
        }else{
            current->getPrev()->setNext(current->getNext());
        }
        if(current == last) {
            NodoCliente *eliminado = last;
            last = last->getPrev();
            delete eliminado;
        }else{
            current->getNext()->setPrev(current->getPrev());
        }
        return true;
    }else{
        return false;
    }

}


ListaClientes::~ListaClientes() {
}

